app/scripts/views/settings.js

```diff
@@ -148,6 +149,34 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
       return self._showAvatar();
     },
 
+    beforeDestroy: function () {
+      $('body.settings').fadeOut(this.FADE_OUT_SETTINGS_MS, function (){
+        $('body').removeClass('settings').show();
+      });
+    },
+
+    // When the user adds, removes or changes a display name
+    // this gets called and swaps out headers to reflect
+    // the updated state of the account
+    _swapDisplayName: function () {
+      var account = this.getSignedInAccount();
+      var displayName = account.get('displayName');
+      var email = account.get('email');
+
+      var cardHeader = this.$('.card-header');
+      var cardSubheader = this.$('.card-subheader');
+
+      if (displayName === cardHeader.text()) {
+        return;
+      } else if (! displayName) {
+        cardHeader.text(email);
+        cardSubheader.text('');
+      } else {
+        cardHeader.text(displayName);
+        cardSubheader.text(email);
+      }
+    },
```

Comment:
above is not covered by unit tests, see 
![](http://v14d.com/i/56055bf0da3c4.jpg)

https://coveralls.io/builds/3670185/source?filename=app%2Fscripts%2Fviews%2Fsettings.js

Let us know if you need help adding tests for these